x/hard/keeper/borrow_test.go

```diff
@@ -25,268 +24,515 @@ const (
+					expectedModAccountBalance: sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(1080*KAVA_CF)), sdk.NewCoin("usdx", sdkmath.NewInt(200*USDX_CF)), sdk.NewCoin("busd", sdkmath.NewInt(100*BUSD_CF))),
+
+					expectPass: true,
+				},
+				{
+					expectedAccountBalance:    sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(40*KAVA_CF)), sdk.NewCoin("btcb", sdkmath.NewInt(100*BTCB_CF)), sdk.NewCoin("bnb", sdkmath.NewInt(100*BNB_CF)), sdk.NewCoin("xyz", sdkmath.NewInt(1))),
+					expectedModAccountBalance: sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(1060*KAVA_CF)), sdk.NewCoin("usdx", sdkmath.NewInt(200*USDX_CF)), sdk.NewCoin("busd", sdkmath.NewInt(100*BUSD_CF))),
+					expectPass:                true,
+				},
+			},
+			borrows: []borrowArgs{
+				{
+					borrowCoins: sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(20*KAVA_CF))),
+					wantErr:     "",
+				},
+				{
+					borrowCoins: sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(20*KAVA_CF))),
+					wantErr:     "",
+				},
+			},
+		},
+		{
+			name: "first valid borrow, second insufficient funds blocks",
+			setup: setupArgs{
+				usdxBorrowLimit:    sdk.MustNewDecFromStr("100000000000"),
+				priceKAVA:          sdk.MustNewDecFromStr("5.00"),
+				loanToValueKAVA:    sdk.MustNewDecFromStr("0.6"),
+				priceBTCB:          sdk.MustNewDecFromStr("0.00"),
+				loanToValueBTCB:    sdk.MustNewDecFromStr("0.01"),
+				priceBNB:           sdk.MustNewDecFromStr("0.00"),
+				loanToValueBNB:     sdk.MustNewDecFromStr("0.01"),
+				borrower:           sdk.AccAddress(crypto.AddressHash([]byte("test"))),
+				depositCoins:       []sdk.Coin{sdk.NewCoin("ukava", sdkmath.NewInt(100*KAVA_CF))},
+				initialBorrowCoins: sdk.NewCoins(),
+			},
+			expected: []expected{
+				{
+					expectedAccountBalance:    sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(20*KAVA_CF)), sdk.NewCoin("btcb", sdkmath.NewInt(100*BTCB_CF)), sdk.NewCoin("bnb", sdkmath.NewInt(100*BNB_CF)), sdk.NewCoin("xyz", sdkmath.NewInt(1))),
+					expectedModAccountBalance: sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(1080*KAVA_CF)), sdk.NewCoin("usdx", sdkmath.NewInt(200*USDX_CF)), sdk.NewCoin("busd", sdkmath.NewInt(100*BUSD_CF))),
+
+					expectPass: true,
+				},
+				{
+					expectedAccountBalance:    sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(40*KAVA_CF)), sdk.NewCoin("btcb", sdkmath.NewInt(100*BTCB_CF)), sdk.NewCoin("bnb", sdkmath.NewInt(100*BNB_CF)), sdk.NewCoin("xyz", sdkmath.NewInt(1))),
+					expectedModAccountBalance: sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(1060*KAVA_CF)), sdk.NewCoin("usdx", sdkmath.NewInt(200*USDX_CF)), sdk.NewCoin("busd", sdkmath.NewInt(100*BUSD_CF))),
+					expectPass:                true,
+				},
+			},
+			borrows: []borrowArgs{
+				{
+					borrowCoins: sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(20*KAVA_CF))),
+					wantErr:     "",
+				},
+				{
+					borrowCoins: sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(20*KAVA_CF))),
+					wantErr:     "",
```

Comment:
Is this second borrow supposed to fail with "not enough collateral supplied by account"?

With the test name "... second insufficient funds blocks"

Reply:
I thought I removed this one already since it's already being handled above in multiple test cases. good catch

Reply:
[done](https://github.com/Kava-Labs/kava/pull/1981/commits/59624f31ff71e3c7aa37aca411f6fbe874b9d55a)